concent_api/core/transfer_operations.py

```diff
@@ -107,95 +107,92 @@ def store_pending_message(
     return _create_file_transfer_token(
-        report_computed_task,
-        authorized_client_public_key,
-        operation,
-        should_add_source=should_add_source,
+        subtask_id=subtask_id,
+        source_package_path=get_storage_source_file_path(task_id, subtask_id),
+        source_size=report_computed_task.task_to_compute.size,
+        source_package_hash=report_computed_task.task_to_compute.package_hash,
+        result_package_path=get_storage_result_file_path(task_id, subtask_id),
+        result_size=report_computed_task.size,
+        result_package_hash=report_computed_task.package_hash,
+        authorized_client_public_key=authorized_client_public_key,
+        operation=operation,
+        token_expiration_deadline=calculate_token_expiration_deadline(operation, report_computed_task)
     )
 
 
 def _create_file_transfer_token(
-    report_computed_task: message.tasks.ReportComputedTask,
-    authorized_client_public_key: bytes,
-    operation: FileTransferToken.Operation,
-    deadline: Optional[int] = None,
-    should_add_source: bool = False,
+    subtask_id: str = None,
+    source_package_path: str = None,
+    source_size: int = None,
+    source_package_hash: str = None,
+    result_package_path: str = None,
+    result_size: int = None,
+    result_package_hash: str = None,
+    authorized_client_public_key: bytes = None,
+    operation: FileTransferToken.Operation = None,
+    token_expiration_deadline: Optional[int] = None,
 ) -> FileTransferToken:
-    """
-    Function to create FileTransferToken from ReportComputedTask message
-    """
-    def calculate_token_expiration_deadline(
-        operation: FileTransferToken.Operation,
-        deadline: Optional[int],
-    ) -> int:
-        if deadline is not None:
-            token_expiration_deadline = get_current_utc_timestamp() + deadline
-        else:
-            if operation == FileTransferToken.Operation.upload:
-                token_expiration_deadline = (
-                        report_computed_task.task_to_compute.compute_task_def['deadline'] +
-                        3 * settings.CONCENT_MESSAGING_TIME +
-                        2 * calculate_maximum_download_time(report_computed_task.size)
-                )
-
-            elif operation == FileTransferToken.Operation.download:
-                token_expiration_deadline = (
-                        report_computed_task.task_to_compute.compute_task_def['deadline'] +
-                        calculate_subtask_verification_time(report_computed_task)
-                )
-        return token_expiration_deadline
-
-    task_id         = report_computed_task.task_to_compute.compute_task_def['task_id']
-    subtask_id      = report_computed_task.task_to_compute.compute_task_def['subtask_id']
-    file_path       = get_storage_result_file_path(task_id, subtask_id)
 
+    assert (source_size and source_package_hash) or (result_size and result_package_hash)
```

Comment:
Shouldn't it also check `result_package_path` and `source_package_path` ?

Reply:
Done

Reply:
These are ints and strings, not conditional expressions. Please use full comparisons.

Are you sure that `source_size` == 0 and `result_size` == 0 can never happen?